torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1111,7 +1112,9 @@ def _get_param_id_to_param_from_optim_input(
 
 def _get_flat_param_to_fqn(model: torch.nn.Module) -> Dict[nn.Parameter, str]:
     def module_fn(module, prefix, tree_level, flat_param_to_fqn):
-        for param_name, param in module.named_parameters(recurse=False):
+        for param_name, param in _named_parameters_with_duplicates(
+            module, remove_duplicate=False
```

Comment:
nit: "remove_duplicate=False" ==> "recurse=False"

Reply:
@fegin Is it not possible for there to be a duplicate with `recurse=False`? Is it okay that we are now using `recurse=True` with `remove_duplicate=False`?

I am wondering when we should use `_named_parameters_with_duplicates` with `remove_duplicate=False` and `recurse=True` (like here) versus `_named_parameters_with_duplicates` with just `recurse=False` (like `_init_flat_param_metadata` in flat_param.py).

Reply:
Good point. I added a check to avoid users using `remove_duplicate` with `_named_parameters_with_duplicates`. For `_get_flat_param_to_fqn`, `recurse` does not seem to affect the correctness but only redundant processes. So there are no UT errors. Added the `recurse=True`.